### PR TITLE
163 hoc component to check login

### PR DESCRIPTION
**Pull request by SorochynskyiVladyslav**

### CODE REVIEW

File: src/client/components/auth-hoc/index.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    componentWillMount() {
+        this.props.getCurrentUser();
+    }
+
+    render() {
+        const { component, user } = this.props;
+        if (!user) {
+            return null;
+        }
+        return new component().render();
```
Comment (undefeat) [inline src/client/components/auth-hoc/index.jsx]:
```javascript
return <component />
```

File: src/client/components/auth-hoc/index.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+class AuthHOC extends React.Component {
+    componentWillMount() {
+        this.props.getCurrentUser();
+    }
+
+    render() {
+        const { component, user } = this.props;
+        if (!user) {
+            return null;
```
Comment (undefeat) [inline src/client/components/auth-hoc/index.jsx]: It's better if it shows "Please log in to see the page" with a Login button.

File: src/client/components/auth-hoc/index.jsx (hunk 1 above)
Comment (SorochynskyiVladyslav) [reply src/client/components/auth-hoc/index.jsx]: done

File: src/client/components/auth-hoc/index.jsx (hunk 2 above)
Comment (SorochynskyiVladyslav) [reply src/client/components/auth-hoc/index.jsx]: it redirects to login, when userService.getCurrentUser finds nothing

**Pull request merged**